test/ExecutionFramework.cpp

```diff
@@ -174,29 +175,36 @@ void ExecutionFramework::sendMessage(bytes const& _data, bool _isCreation, u256
 	if (_isCreation)
 	{
 		message.kind = EVMC_CREATE;
-		message.destination = EVMHost::convertToEVMC(h160{});
+		message.recipient = {};
+		message.code_address = {};
 	}
 	else
 	{
 		message.kind = EVMC_CALL;
-		message.destination = EVMHost::convertToEVMC(m_contractAddress);
+		message.recipient = EVMHost::convertToEVMC(m_contractAddress);
+		message.code_address = message.recipient;
 	}
+
 	message.gas = InitialGas.convert_to<int64_t>();
 
-	evmc::result result = m_evmcHost->call(message);
+	evmc::Result result = m_evmcHost->call(message);
 
 	m_output = bytes(result.output_data, result.output_data + result.output_size);
 	if (_isCreation)
 		m_contractAddress = EVMHost::convertFromEVMC(result.create_address);
 
 	m_gasUsed = InitialGas - result.gas_left;
+	unsigned const refundRatio = (m_evmVersion >= langutil::EVMVersion::london() ? 5 : 2);
```

Comment:
This change adds support for calculating refunds, which was not the case prior.

Reply:
Can we decouple this change from the actual upgrade? I mean, we should definitely fix this, but mixing it together here makes it rather unclear what causes what in the PR.

Reply:
It is a separate commit, so already decoupled, the differences exists before and after it.

Reply:
Moved to #13695.